visualizer.js

```diff
@@ -0,0 +1,302 @@
+const screenWidth = () => window.innerWidth
+const screenHeight = () => window.innerHeight
+
+let catPos = []
+let datPos = []
+
+//state or mode
+const [ CATEGORY_MODE, ORGS_MODE ] = [ 'CATEGORY_MODE', 'ORGS_MODE']
+
+const data = []
+const category = []
+
+const randomPos = (centrex, centrey, radius, tolerance) => {
+  return [
+    centrex + random(-radius, radius) * tolerance,
+    centrey + random(-radius, radius) * tolerance 
+  ]
+}
+
+
+loadData(json => {
+  json.forEach(element => {
+    data.push(element)
+    if(category.find(value => value.category_name == element.category) == null)
+      category.push({
+        category_name : element.category,
+        bubble_size : 0,
+        frequency : 1
+      })
+    else
+      category.find(value => value.category_name == element.category).frequency++
+
+  })
+  category.sort((a,b) => b.frequency - a.frequency).forEach((e, index) => {
+    
+    e.bubble_size = (screenWidth()/3 * e.frequency/212)
+
+    
+
+    //need a better way to randomize bubble position without
+    //any overlaps between bubble    
+    let newPos = [
+      random(-screenWidth() * 0.4, screenWidth() * 0.4),
+      random(-screenHeight() * 0.4, screenHeight() * 0.4)
+    ]
+
+    if(index > 0){
+
+      for(let j = index - 1; j >= 0; j--)
+        while(dist(...newPos, ...catPos[j]) <  2 * category[j].bubble_size)
+          newPos = [
+            random(-screenWidth() * 0.4, screenWidth() * 0.4),
+            random(-screenHeight() * 0.4, screenHeight() * 0.4)
+          ]
+
+
+    }
+        
+    catPos.push(newPos)
+  
+  })
+})
+
+let rot = 0 
+let trig = [false, 0]
+let currState = CATEGORY_MODE
+let selectedCategory = category[0]
+let node = []
+
+const view = [
+  screenWidth()/2, 
+  screenHeight()/2
+]
+
+const getcoord = (x, y) => {
+  return [x - screenWidth()/2, y - screenHeight()/2]
```

Comment:
`const getcoord = (x, y) => [x - screenWidth()/2, y - screenHeight()/2]` is equivalence.